src/encoded/ingestion/loadxl_extensions.py

```diff
@@ -0,0 +1,75 @@
+            # SKIP: feedbeef-eb17-4406-adb8-060ea2ae2180
+            # CHECK: cafebabe-eb17-4406-adb8-0eacafebabe
+            # ERROR: deadbeef-483e-4a08-96b9-3ce85ce8bf8c
+            # Note that SKIP means skip POST (create); it still may do PATCH (update), if overwrite.
+            if isinstance(item, bytes):
+                item = item.decode("ascii")
+            elif not isinstance(item, str):
+                continue
+            match = LOADXL_RESPONSE_PATTERN.match(item)
+            if not match or match.re.groups != 2:
+                continue
+            action = LOADXL_ACTION_NAME[match.group(1).upper()]
+            identifying_value = match.group(2)
+            if not response.get(action):
+                response[action] = []
+            response[action].append(identifying_value)
+        # Items flagged as SKIP in loadxl could ultimately be a PATCH (update),
+        # so remove from the skip list any items which are also in the update list. 
+        response["skipped"] = [item for item in response["skipped"] if item not in response["updated"]]
+        # Items flagged as POST (create) in loadxl typically also are flagged as PATCH (update), due to the
+        # way they are written, so remove from the update list any items which are also in the create list.
+        response["updated"] = [item for item in response["updated"] if item not in response["created"]]
+        response["types"] = list(data.keys())
+        return response
+
+    loadxl_load_data_response = loadxl_load_data(
+        testapp=portal_vapp,
+        inserts=data,
+        docsdir=None,
+        overwrite=True,
+        itype=None,
+        from_json=True,
+        patch_only=False,
+        validate_only=validate_only)
+
+    return package_loadxl_response(loadxl_load_data_response)
+
+
+def summarize_load_data_into_database_response(load_data_response: Optional[dict],
+                                               submission: SmahtSubmissionFolio) -> list[str]:
+    """
+    Summarize the given load data results into a simple short list of English phrases;
+    this will end up going into the additional_properties of the IngestionSubmission
+    object in the Portal database (see SubmissionFolio.record_results); this is what will
+    get displayed, by default, by the submitr tool when it detects processing has completed.
+    """
+    return [
+        f"Successful ingestion summary:",
+        f"In File: {submission.data_file_name}",
+        f"S3 File: {submission.s3_data_file_location}",
+        f"Details: {submission.s3_details_location}",
+        f"N Types: {len(load_data_response['types'])}",
+        f"Created: {len(load_data_response['created'])}",
+        f"Updated: {len(load_data_response['updated'])}",
+        f"Skipped: {len(load_data_response['skipped'])}",
+        f"Checked: {len(load_data_response['validated'])}",
+        f"Errored: {len(load_data_response['errors'])}"
```

Comment:
Does it happen that we get errors in the middle even with stuff that passes the schema? That could be really messy.

Reply:
Not entirely sure what mean - the only kinds of errors we could get for an item that passes the schema validation are errors during actual loading, which are collected/itemized in this module.